hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -1212,3 +1212,44 @@ def test_basic_indices_generate_valid_indexers(
 def test_array_owns_memory(x: np.ndarray):
     assert x.base is None
     assert x[...].base is x
+
+
+@given(nps.arrays(nps.floating_dtypes(), nps.array_shapes()))
+def test_make_noncontiguous(array):
+    np.testing.assert_array_equal(array, nps.make_noncontiguous(array))
+
+
+@given(nps.arrays(nps.floating_dtypes(), nps.array_shapes()), st.data())
+def test_permute_dimensions(array, data):
+    permutation = data.draw(st.permutations(range(array.ndim)))
+    np.testing.assert_array_equal(array, nps.permute_dimensions(array, permutation))
+
+
+@given(nps.arrays(nps.floating_dtypes(), nps.array_shapes()), st.data())
+def test_permute_dimensions_explicit(array, data):
+    perm = data.draw(st.permutations(range(array.ndim)))
+    np.testing.assert_array_equal(
+        array, nps.permute_dimensions(array, permuted_indices=perm)
+    )
+
+
+@given(
+    nps.array_memory_scramblers(), nps.arrays(nps.floating_dtypes(), nps.array_shapes())
+)
+def test_array_memory_scramblers(scrambler, array):
+    scrambled = scrambler(array)
+    # A scrambler should always return an equivalent array
+    np.testing.assert_array_equal(array, scrambled)
+
```

Comment:
Add a test that uses `tests.common.debug.minimal` to test that the returned scramblers shrink to the identity function.